tests/lib.rs

```diff
@@ -138,6 +138,89 @@ async fn new_read() -> Result<(), SelfEncryptionError> {
     Ok(())
 }
 
+#[tokio::test]
+async fn write_and_close_random_sizes_at_random_positions() -> Result<(), SelfEncryptionError> {
+    let mut rng = new_test_rng()?;
+    let mut storage = SimpleStorage::new();
+    let max_broken_size = 20 * 1024;
+    let original = random_bytes(&mut rng, DATA_SIZE as usize);
+    // estimate number of broken pieces, not known in advance
+    let mut broken_data: Vec<(u32, &[u8])> =
+        Vec::with_capacity((DATA_SIZE / max_broken_size) as usize);
+
+    let mut offset = 0;
+    let mut last_piece = 0;
+    while offset < DATA_SIZE {
+        let size;
+        if DATA_SIZE - offset < max_broken_size {
+            size = DATA_SIZE - offset;
+            last_piece = offset;
+        } else {
+            size = rand::random::<u32>() % max_broken_size;
+        }
+        let piece: (u32, &[u8]) = (offset, &original[offset as usize..(offset + size) as usize]);
+        broken_data.push(piece);
+        offset += size;
+    }
+
+    {
+        let slice_broken_data = &mut broken_data[..];
+        slice_broken_data.shuffle(&mut rng);
+    }
+
+    match broken_data.iter().filter(|&x| x.0 != last_piece).last() {
+        None => panic!("Should never occur. Error in test itself."),
+        Some(overlap) => {
+            let mut extra: Vec<u8> = overlap.1.to_vec();
+            extra.extend(random_bytes(&mut rng, 7usize)[..].iter().cloned());
```

Comment:
define a const for this `majic` number

Reply:
This code is just copied pasted from the previous function, just that we close after every write.